src/module.c

```diff
@@ -8327,18 +8351,18 @@ void ModuleForkDoneHandler(int exitcode, int bysignal) {
  *         int32_t dbnum_first;    // Swap Db first dbnum
  *         int32_t dbnum_second;   // Swap Db second dbnum
  *
- * * RedisModuleEvent_ReplBackup
+ * * RedisModuleEvent_ReplAsync
  *
- *     Called when diskless-repl-load config is set to swapdb,
- *     And redis needs to backup the the current database for the
- *     possibility to be restored later. A module with global data and
- *     maybe with aux_load and aux_save callbacks may need to use this
- *     notification to backup / restore / discard its globals.
```

Comment:
@yossigo if we deprecate `RedisModuleEvent_ReplAsync`, we surely still wanna keep it in the header file, so old modules can still handle it (although their code that handles it is now dead code) when used on new redis.
but what about the documentation? do we wanna drop the old API from the documentation, or keep it here with a deprecation notice?

Reply:
I think we need to keep the documentation and mention it is deprecated, as well as use `__attribute__((deprecated))` to produce a compile time warning.

Reply:
ok. @eduardobr, please revive the doc comment and add the deprecation notice and attribute

Reply:
Change made, but I couldn't identify based on current code how to precisely apply the deprecated attribute in some cases. For example, before or after the elements?

Reply:
i'm not sure myself. i suggest you try to revive the test code that uses them and see that you get the warning.
i guess at least one warning is ok too (if we can't get them all to work).

Reply:
In current state, it will generate this warning on ./runtest-moduleapi
![image](https://user-images.githubusercontent.com/3770445/134970494-e6c49ec6-559f-4081-bddf-8824fa391041.png)
Using the attribute on REDISMODULE_SUBEVENT_REPL_BACKUP_* variables wouldn't work.
Please let me know if that's the desired outcome of the deprecation overall

Reply:
i think one warning is enough.
and i think we should convert the test in `tests/modules/testrdb.c` to use the new API.
no sense in keeping the dead code anyway (apart from getting the annoying deprecation warning).

Reply:
Just one thing before I commit. Is the goal of testrdb.c:replBackupCallback to check if the order of events is correct?
We could use some comments to understand this file a bit better.

Reply:
i think it was attempting to use the callback for the purpose for which they were added, and see that they work correctly.
i.e. it's a module with global data, that's stored into the aux fields of the rdb, and it was using the callbacks to create a backup and restore / discard it.

similarly i guess the new test should use the new apis, and check that during rdb loading the old data is still accessible, and that if loading fails the temp data (that's already been loaded) is freed correctly, and that it the loading succeeds, the new data is accessible and the old one is released.